File: tools/parity/dispatch/fetch-error-classification.ts
// TS dispatch for the fetch-error-classification parity module: maps the shared
// vector function names to the real `src/main/git/fetch-error-classification.ts`
// exports so the harness compares the live TS reference against the Rust port.

import { isMissingRemoteRefGitError } from '../../../src/main/git/fetch-error-classification'

export function dispatch(fn: string, input: unknown): unknown {
  switch (fn) {
    case 'isMissingRemoteRefGitError':
      // Vectors carry the error message string directly; the function does
      // String(error) for non-Error inputs, so a string is an exact stand-in.
      return isMissingRemoteRefGitError(input)
    default:
      throw new Error(`unknown function ${fn}`)
  }
}

File: rust/crates/orca-parity/src/modules/fetch_error_classification.rs
//! Parity dispatch for `orca_git::fetch_error_classification` vs
//! `src/main/git/fetch-error-classification.ts`.

use orca_git::fetch_error_classification::is_missing_remote_ref_git_error;
use serde_json::{json, Value};

pub fn dispatch(function: &str, input: &Value) -> Value {
    match function {
        "isMissingRemoteRefGitError" => {
            // TS resolves the message via String(error); vectors carry it directly.
            let message = input.as_str().unwrap_or_default();
            Value::Bool(is_missing_remote_ref_git_error(message))
        }
        other => json!({ "__parity_error__": format!("unknown function {other}") }),
    }
}
